tutorials/hardhat-tutorial.md

```diff
@@ -0,0 +1,425 @@
+    
+    // To deploy our StakingDao, you need to call delegationDao.deploy() and await
+    // its deployed() method, which happens once its transaction has been mined.
+    const deployedDao = await delegationDao.deploy(targetCollator, deployer.address);
+
+    await deployedDao.deployed();
+
+    // Fixtures can return anything you consider useful for your tests
+    return { deployedDao };
+  }
+
+    async function deployDaoFixtureWithMembers() {
+    const [deployer, member1] = await ethers.getSigners();
+
+    const delegationDao = await ethers.getContractFactory("DelegationDAO");
+    const deployedDao = await delegationDao.deploy(targetCollator, deployer.address);
+
+    await deployedDao.deployed();
+    await deployedDao.grant_member(member1.address);
+
+    return { deployedDao};
+  }
+```
+
+### Writing your First Test Cases {: #writing-your-first-test-cases }
+
+This tutorial deviates from the [test driven development philosophy](https://en.wikipedia.org/wiki/Test-driven_development){target=_blank} of writing your tests prior to your code. A few test cases will be cherry picked for demonstration purposes. 
+
+First, you'll create a subsection called `deployment` to keep the test file organized. This isn't required but may be helpful for organization purposes. Next you'll define your first test case by using the `it` mocha function. A description that clearly indicates what the test is doing is provided as well as indication that the function is async. This first test is simply checking to see that the StakingDAO is correctly storing the address of the target collator.
+
+Next, a fixture is loaded - in this case, the "empty" stakingDAO fixture. Fixtures are easily interchangeable within your test cases. 
+
+```javascript
+// You can nest describe calls to create subsections.
+describe("Deployment", function () {
+    // `it` is another Mocha function. This is the one you use to define each
+    // of your tests. It receives the test name, and a callback function.
+    //
+    // If the callback function is async, Mocha will `await` it.
+    it("Should have correct target collator", async function () {
+      // We use loadFixture to setup our environment, and then assert that
+      // things went well
+      const { deployedDao, deployer } = await loadFixture(deployDaoFixture);
+
+      // `expect` receives a value and wraps it in an assertion object. These
+      // objects have a lot of utility methods to assert values.
+      expect(await deployedDao.target()).to.equal(targetCollator);
+});
+```
+
+Let's add another test case. When a StakingDAO is launched, it shouldn't have any funds. This test verifies that is indeed the case. Go ahead and add the following test case to your `Dao.js` file:
+
+```javascript
+it("The DAO should initially have 0 funds in it", async function () {
+	const { deployedDao, deployer } = await loadFixture(deployDaoFixture);
+	expect(await deployedDao.totalStake()).to.equal(0);
+});
+```
```

Comment:
I ran these tests and they both failed. So I tried to look up `loadFixture` in hardhats docs and it only shows up on the hardhat-waffle plugin page so I am a bit confused: https://hardhat.org/hardhat-runner/plugins/nomiclabs-hardhat-waffle#environment-extensions

The error I got when running the tests:

```
     OnlyHardhatNetworkError: This helper can only be used with Hardhat Network. You are connected to 'moonbase'.
      at checkIfHardhatNetwork (/Users/es/Documents/work/kevin-hardhat-tutoral/hardhat/node_modules/@nomicfoundation/hardhat-network-helpers/src/utils.ts:28:11)
      at getHardhatProvider (/Users/es/Documents/work/kevin-hardhat-tutoral/hardhat/node_modules/@nomicfoundation/hardhat-network-helpers/src/utils.ts:39:9)
      at takeSnapshot (/Users/es/Documents/work/kevin-hardhat-tutoral/hardhat/node_modules/@nomicfoundation/hardhat-network-helpers/src/helpers/takeSnapshot.ts:20:20)
      at loadFixture (/Users/es/Documents/work/kevin-hardhat-tutoral/hardhat/node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:60:22)
      at Context.<anonymous> (/Users/es/Documents/work/kevin-hardhat-tutoral/hardhat/tests/Dao.js:65:47)
      ```

Reply:
My bad, yup fixtures are only for hardhat network so I removed any references to fixtures.